blob/fileblob/fileblob.go

```diff
@@ -91,7 +102,13 @@ func openBucket(dir string, _ *Options) (driver.Bucket, error) {
 	if !info.IsDir() {
 		return nil, fmt.Errorf("%s is not a directory", dir)
 	}
-	return &bucket{dir}, nil
+	b := &bucket{dir: dir}
```

Comment:
I don't think it's going to be possible to support `SignedURL` if the user hasn't provided `opts.URLSigner`. You could conceivably provide a default for the signing part, but if they haven't provided a hook for HTTP requests, the URLs won't work.

How about this:

```
if opts == nil {
  opts = &Options{}
}
return &bucket{dir: dir, opts: opts}
```

(and update `bucket` to have `opts *Options` instead of the `URLSigner` explicitly).

Reply:
Can confirm, was not planning on trying to support `SignedURL` if no `opts.URLSigner` was supplied. 

The plan was to leave `errorURLSigner` as the default, which would have no impact on users who don't intend or attempt to use signed urls, though I could give `errorURLSigner` a more informative message to the effect of 'you need to supply a URLSigner if you want to use this', as right now it just returns an error with the message "not implemented"?